setup.py

```diff
@@ -58,7 +58,10 @@
         "gitpython",
         "psutil",
     ],
-    extras_require={"reports": ["jinja2", "networkx", "pygments", "pygraphviz"]},
+    extras_require={
+        "reports": ["jinja2", "networkx", "pygments", "pygraphviz"],
+        "jupyter": ["jupyter", "jupyter_contrib_nbextensions"]
```

Comment:
I think it should be enough to depend on nbformat. For running the notebook rule, one simply has to define a conda or singularity directive which includes jupyter and whatever other packages are needed for running the notebook.

Reply:
I agree, I think this should be mentioned in the documentation though for new users.

Reply:
That seems sensible. Should `nbformat` be added to `install_requires`?

Reply:
Yes, I think so. Once this is available, many workflows will make use of it, because it is such a cool feature :-).

Reply:
Great, I amended the requirements!